rasa/core/policies/ensemble.py

```
@@ -302,18 +304,72 @@ def _ensure_loaded_policy(cls, policy, policy_cls, policy_name: Text):
             policy_cls = registry.policy_from_module_path(policy_name)
             dir_name = f"policy_{i}_{policy_cls.__name__}"
             policy_path = os.path.join(path, dir_name)
-            policy = policy_cls.load(policy_path)
+
+            if new_config:
+                if "should_finetune" not in rasa.shared.utils.common.arguments_of(
+                    policy_cls.load
+                ):
+                    raise UnsupportedDialogueModelError(
+                        f"{policy_cls.__name__} does not support fine-tuning. "
+                        f"To support fine-tuning the `load` method must have the "
+                        f"argument `should_finetune`. This argument should be added to "
+                        f"all policies by Rasa Open Source 3.0.0."
+                    )
+
+                config_for_policy = new_config["policies"][i]
+                epochs = cls._get_updated_epochs(
+                    policy_cls, config_for_policy, finetuning_epoch_fraction
+                )
+                if epochs:
+                    if "epoch_override" not in rasa.shared.utils.common.arguments_of(
+                        policy_cls.load
+                    ):
+                        raise UnsupportedDialogueModelError(
+                            f"{policy_cls.__name__} does not support fine-tuning. "
+                            f"To support fine-tuning the `load` method must have the "
+                            f"argument `epoch_override` if the policy uses epochs."
+                            f"This argument should be added to all policies by "
+                            f"Rasa Open Source 3.0.0."
+                        )
+
+                    policy = policy_cls.load(
+                        policy_path, should_finetune=True, epoch_override=epochs
+                    )
+                else:
+                    policy = policy_cls.load(policy_path, should_finetune=True)
```

I guess this should also be covered in the above `if` construct which should throw a warning for unsupported policy otherwise for custom policies with custom method for `load`, this line will error out?